### PR TITLE
network: several interface up/down and setting MTU related updates

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/network/networkd-setlink.c [hunk 1]
```diff
@@ -677,16 +677,38 @@ int link_request_to_set_master(Link *link) {
...
+        }
 
-        /* IPv6 protocol requires a minimum MTU of IPV6_MTU_MIN(1280) bytes on the interface. Bump up
-         * MTU bytes to IPV6_MTU_MIN. */
-        if (mtu < IPV6_MIN_MTU && link_ipv6_enabled(link)) {
-                log_link_warning(link, "Bumping MTU to " STRINGIFY(IPV6_MIN_MTU) ", as IPv6 is enabled "
-                                 "and requires a minimum MTU of " STRINGIFY(IPV6_MIN_MTU) " bytes");
-                mtu = IPV6_MIN_MTU;
+        if (mtu > link->max_mtu) {
+                log_link_warning(link, "Reduces the requested MTU %"PRIu32" to the interface's maximum MTU %"PRIu32".",
```
Comment (poettering) [inline src/network/networkd-setlink.c]: Reduces → Reducing?

File: src/network/networkd-sysctl.c [hunk 2]
```diff
@@ -199,7 +201,14 @@ int link_set_ipv6_mtu(Link *link) {
         if (link->network->ipv6_mtu == 0)
                 return 0;
 
-        return sysctl_write_ip_property_uint32(AF_INET6, link->ifname, "mtu", link->network->ipv6_mtu);
+        mtu = link->network->ipv6_mtu;
+        if (mtu > link->max_mtu) {
+                log_link_warning(link, "Reduces requested IPv6 MTU %"PRIu32" to the interface's maximum MTU %"PRIu32".",
```
Comment (poettering) [inline src/network/networkd-sysctl.c]: ditto

File: src/network/networkd-link.c [hunk 3]
```diff
@@ -708,9 +708,9 @@ int link_handle_bound_to_list(Link *link) {
                 }
 
         if (!required_up && link_is_up)
-                return link_down(link);
+                return link_request_to_bring_up_or_down(link, false);
```
Comment (poettering) [inline src/network/networkd-link.c]: doesn't matter, but maybe insert `/* up= */` before the `false`

File: src/network/networkd-link.c [hunk 4]
```diff
@@ -708,9 +708,9 @@ int link_handle_bound_to_list(Link *link) {
...
         if (required_up && !link_is_up)
-                return link_up(link);
+                return link_request_to_bring_up_or_down(link, true);
```
Comment (poettering) [inline src/network/networkd-link.c]: similar

File: src/network/networkd-link.c [hunk 5]
```diff
@@ -1581,8 +1581,8 @@ static int link_admin_state_up(Link *link) {
                 return 0;
 
         if (link->activated && link->network->activation_policy == ACTIVATION_POLICY_ALWAYS_DOWN) {
-                log_link_info(link, "ActivationPolicy is \"always-off\", forcing link down");
-                return link_down(link);
+                log_link_info(link, "ActivationPolicy is \"always-off\", forcing link down.");
+                return link_request_to_bring_up_or_down(link, false);
```
Comment (poettering) [inline src/network/networkd-link.c]: ditto

File: src/network/networkd-setlink.c [hunk 6]
```diff
@@ -103,35 +103,35 @@ static int link_set_addrgen_mode_handler(sd_netlink *rtnl, sd_netlink_message *m
 }
 
 static int link_set_bond_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *link) {
-        return set_link_handler_internal(rtnl, m, link, SET_LINK_BOND, true, NULL);
+        return set_link_handler_internal(rtnl, m, link, SET_LINK_BOND, false, NULL);
```
Comment (poettering) [inline src/network/networkd-setlink.c:123]: ditto

**Comment by yuwata** [comment]
I think this is ready for review. Please take a look.

File: src/network/networkd-setlink.h [hunk 7]
```diff
@@ -11,6 +11,7 @@ typedef enum SetLinkOperation {
         SET_LINK_BOND,                    /* Setting bond configs. */
         SET_LINK_BRIDGE,                  /* Setting bridge configs. */
         SET_LINK_BRIDGE_VLAN,             /* Setting bridge vlan configs. */
+        SET_LINK_CAN,                     /* Setting can interface configs. */
```
Comment (poettering) [inline src/network/networkd-setlink.h]: given that CAN is an acronym, maybe uppercase it? (same for "vlan" in the line prior)

File: src/network/networkd-setlink.c (hunk 6 above)
Comment (poettering) [reply src/network/networkd-setlink.c:123]: still would be nice

**Comment by yuwata** [comment]
@poettering Thank you for your review. The above point is addressed. Upgrading the green label.

**Pull request merged**